src/tests/ut/reserve_queue/reserve_queue_test.cpp

```diff
@@ -0,0 +1,412 @@
+#include "reserve_queue_test.hpp"
+
+#include "gmock/gmock.h"
+
+namespace
+{
+using ::testing::ElementsAre;
+using ::testing::UnorderedElementsAre;
+
+class AddAwaitingPatientShould : public ReserveQueueFixture
+{
+};
+
+TEST_F(AddAwaitingPatientShould, EnqueuePatientsForTreatmentInCorrectOrder)
+{
+    Treatment filling_treatment{Treatment::DentalFilling};
+    sut_.addAwaitingPatient(filling_treatment, patients_[0]);
+    sut_.addAwaitingPatient(filling_treatment, patients_[1]);
+    sut_.addAwaitingPatient(filling_treatment, patients_[2]);
+    sut_.addAwaitingPatient(filling_treatment, patients_[3]);
+
+    Treatment tooth_extraction_treatment{Treatment::ToothExtraction};
+    sut_.addAwaitingPatient(tooth_extraction_treatment, patients_[3]);
+    sut_.addAwaitingPatient(tooth_extraction_treatment, patients_[0]);
+    sut_.addAwaitingPatient(tooth_extraction_treatment, patients_[1]);
+
+    const auto filling_treatment_queued_patients = sut_.awaitingPatients(filling_treatment);
+    const auto tooth_extraction_queed_patients = sut_.awaitingPatients(tooth_extraction_treatment);
+
+    EXPECT_THAT(filling_treatment_queued_patients, ElementsAre(patients_[0], patients_[1], patients_[2], patients_[3]));
+    EXPECT_THAT(tooth_extraction_queed_patients, ElementsAre(patients_[3], patients_[0], patients_[1]));
+}
```

Comment:
A bit too much happens here. 

1. Number of patiens could be reduced to 2 
2. First test if single Treatment works fine, then another test with two treatments to make sure that patient is not dropped from both lists
3. Test name could follow Given_When_Then or Given_When_Expect conventions, I find it more structured and precise

Reply:
I see 2. is covered in lines 36-43.

Reply:
Ad. 1. I limited the number of patients to 2 (in all further tests as well, except one - testing removing from the middle where 3 where reasonable number).

Ad. 2. Seeing your further comment I suppose that 2. is no longer problematic (please correct me if I misunderstood).

Ad. 3. I do not agree with this one. IMHO this depends on preferred convention. I like this one better, but I can try to adhere to your proposal if you find it crucial.

Just please have a look on example test print:
![image](https://github.com/coders-school/desktop-business-app/assets/82645568/5830cc1e-7aab-4bd9-a02e-58efe4f4ea1b)
One single line with tested function name before the dot and expected behavior seem very explicit about expectation if you read it as a single sentence. Given_When_... is also fine but when much happens in a test it could be hard to make a good name explaining test in my opionion.

Reply:
Ad. 2 correct 
Ad. 3 That's an interesting approach, I have never seen this one 😵‍💫 
The thing is that we followed some convention in different tests and we should do it here also. 
We can let it be with different conventions, that's not a problem, but keeping some consistency is nice in project